src/main/java/backend/UserService.java

```diff
@@ -0,0 +1,42 @@
+package backend;
+
+import org.springframework.security.oauth2.client.userinfo.DefaultOAuth2UserService;
+import org.springframework.security.oauth2.core.user.OAuth2User;
+import org.springframework.security.oauth2.core.user.DefaultOAuth2User;
+import org.springframework.security.oauth2.client.userinfo.OAuth2UserRequest;
+import org.springframework.security.oauth2.client.userinfo.OAuth2UserService;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.stereotype.Service;
+
+import java.util.Collections;
+
+@Service
+public class UserService implements OAuth2UserService<OAuth2UserRequest, OAuth2User> {
+
+    private final UserRepository userRepository;
+
+    public UserService(UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
+
+    @Override
+    public OAuth2User loadUser(OAuth2UserRequest userRequest) {
+        OAuth2UserService<OAuth2UserRequest, OAuth2User> delegate = new DefaultOAuth2UserService();
+        OAuth2User oAuth2User = delegate.loadUser(userRequest);
+
+        String githubId = oAuth2User.getAttribute("id").toString();
+        String email = oAuth2User.getAttribute("email");
+
+        User user = userRepository.findByGithubId(githubId);
+        if (user == null) {
+            user = new User(null, githubId, email);
+            userRepository.save(user);
+        }
```

Comment:
You could think about writing a serviceFunction findUserById and make a reference here. Probably you are going to use this more than once soon.